.github/workflows/ci.yml

```diff
@@ -607,4 +612,37 @@ jobs:
         run: cargo cache --autoclean-expensive
       - name: Stop sccache
         if: always() && steps.sccache.conclusion == 'success'
-        run: sccache --stop-server || true
\ No newline at end of file
+        run: sccache --stop-server || true
+
+  upload-coverage:
+    runs-on: [self-hosted, 8vcpu-16ram-ubuntu22-namada-x86]
+    container: 
+      image: ghcr.io/heliaxdev/namada-ci:namada-main
```

Comment:
lets not use a custom runner for this, we can use github free runners

Reply:
[542992b](https://github.com/anoma/namada/pull/3704/commits/542992b5193d3a66d410c24a6f23abb0b33a3840)